receipts/handler/handler.go

```diff
@@ -164,8 +166,14 @@ func (r *ReceiptHandler) CacheReceipt(
 		r.eventHandler, int32(len(r.receiptsCache)), int64(auth.Nonce(state, caller)),
 	)
 	if err != nil {
-		return []byte{}, errors.Wrap(err, "receipt not written, returning empty hash")
+		return nil, errors.Wrap(err, "receipt not written, returning empty hash")
+	}
+
+	// use go-ethereum tx hash instead of loom tx hash
```

Comment:
`loom tx hash` -> `non-deterministic tx hash`
Pass the tx hash to `WriteReceipt()` and skip the old hash generation code in there if the `txHash != nil`

Reply:
And move the feature check out to the caller of `CacheReceipt`

Reply:
Done